File: BraintreeDataCollector/src/main/java/com/braintreepayments/api/DataCollectorCallback.java
package com.braintreepayments.api;

import android.content.Context;

import androidx.annotation.Nullable;

/**
 * Callback for receiving result of
 * {@link DataCollector#collectDeviceData(Context, DataCollectorCallback)} and
 * {@link DataCollector#collectDeviceData(Context, String, DataCollectorCallback)} and
 * {@link DataCollector#collectPayPalDeviceData(Context, DataCollectorCallback)}.
 */
public interface DataCollectorCallback {

    /**
     * @param deviceData the device information collected for fraud detection
     * @param error an exception that occurred while fetching device data
     */
    void onResult(@Nullable String deviceData, @Nullable Exception error);
}

File: DataCollector/src/main/java/com/braintreepayments/api/datacollector/DataCollectorCallback.kt
package com.braintreepayments.api.datacollector

/**
 * Callback for receiving result of [DataCollector.collectDeviceData]
 */
fun interface DataCollectorCallback {
    /**
     * @param dataCollectorResult the [DataCollectorResult] of collecting device data
     */
    fun onDataCollectorResult(dataCollectorResult: DataCollectorResult)
}
